### PR TITLE
cancels reconciliation when node config already has final state

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/2993.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -48,7 +48,7 @@ required = [
   name = "github.com/giantswarm/micrologger"
 
 [[constraint]]
-  branch = "master"
+  branch = "fix"
```
Comment (kopiczko) [inline Gopkg.toml]: Friendly remainder.

File: service/controller/v1/resource/node/create.go [hunk 2]
```diff
@@ -31,19 +32,27 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
+	if customObject.Status.HasFinalCondition() {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "node config status already has final state")
+		resourcecanceledcontext.SetCanceled(ctx)
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource for custom object")
+
+		return nil
+	}
```
Comment (kopiczko) [inline service/controller/v1/resource/node/create.go]: I like it being on the top. It's like early return.

Comment (xh3b4sd) [reply service/controller/v1/resource/node/create.go]: It is. Only the type magic is above, which we need to have the type and to the method call.

Comment (kopiczko) [reply service/controller/v1/resource/node/create.go]: Yes it is. I realized. I just expressed my appreciation :)

Comment (xh3b4sd) [reply service/controller/v1/resource/node/create.go]: I see. Thanks buddy. <3

File: Gopkg.toml (hunk 1 above)
Comment (xh3b4sd) [reply Gopkg.toml]: Done.

**Pull request merged**